core/src/main/java/org/geysermc/geyser/session/UpstreamSession.java

```diff
@@ -73,6 +90,7 @@ public void sendPostStartGamePackets() {
 
         BedrockPacket packet;
         while ((packet = postStartGamePackets.poll()) != null) {
+            packet.setClientId(clientId);
```

Comment:
Either this line or the one in queuePostStartGamePacket should be removed as they touch the same packets.

Reply:
Done - removed this line